src/SortableList.js

```diff
@@ -104,7 +105,8 @@ export default class SortableList extends Component {
         rowsLayouts: null,
         order: nextOrder
       });
-
+    } else if (data && nextData && !shallowEqual(data, nextData)) {
+      this.setState({ data: nextData });
```

Comment:
issue with code change here: this won't update the order when rendering if the order is different

Reply:
Are you providing the `order` prop? It works fine for me, but I might not have factored in if you don't provide it. The goal is that the data gets rerendered without the keys changing. This line was the issue for me: https://github.com/gitim/react-native-sortable-list/pull/66/files/e0f5ce66ba855cf43e85a3f19456c0d5f38e67b3#diff-e7d95e98bb0472979c2ad48071012cb9R94.

Reply:
yes, I was providing the `order` prop. What would happen is that the first if statement is false (since the keys of the data were the same), but the second if statement failed (since the actual `shallowEqual` comparison failed).
So this would trigger the state to update just the `data`, even though `order` should be updated to `nextOrder` as well.

Does that make sense?

Reply:
I think so...  is it if you're updating order _and_ data at the same time? whereas right now it's assuming each happen in separate calls?

Reply:
Correct. Use case for me is that when a user throws away past changes, we want to restore the last savepoint's order and data (hence, updating both at once)

Reply:
What if we combined the data and order change into one `if` branch? would that work?

Reply:
Maybe something like this:
```
else if ((data && nextData && !shallowEqual(data, nextData)) || (order && nextOrder && !shallowEqual(order, nextOrder)) {
this.setState({ data: nextData, order: nextOrder });
```